File: api/src/main/java/api/config/WebConfig.java
package api.config;

import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
public class WebConfig implements WebMvcConfigurer {

    @Override
    public void addCorsMappings(CorsRegistry registry) {
        registry.addMapping("/**")
                .allowedMethods("HEAD", "GET", "PUT", "POST", "DELETE", "PATCH")
                .allowedOrigins("*")
                .allowedHeaders("X-Requested-With, X-Auth, Content-Type")
                .maxAge(3600)
                .allowCredentials(true)
                .exposedHeaders("Location")
        ;
    }
}

File: api/src/main/java/api/config/WebConfig.kt
package api.config

import api.converter.LocalDateCustomConverter
import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.CorsRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer
import org.springframework.format.FormatterRegistry



//@Configuration
//class WebConfig : WebMvcConfigurer {
//
//    override fun addCorsMappings(registry: CorsRegistry?) {
//        registry!!.addMapping("/**")
//                .allowedMethods("HEAD", "GET", "PUT", "POST", "DELETE", "PATCH")
//                .allowedOrigins("*")
//                .allowedHeaders("X-Requested-With, X-Auth, Content-Type")
//                .maxAge(3600)
//                .allowCredentials(true)
//                .exposedHeaders("Location")
//    }
//
//    override fun addFormatters(registry: FormatterRegistry) {
//        registry.addConverter(LocalDateCustomConverter())
//        registry.javaClass
//    }
//}
